src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -53,58 +72,131 @@ export class HuiDialogEditCard extends LitElement {
     return this.shadowRoot!.querySelector("hui-yaml-card-preview")!;
   }
 
-  protected render() {
+  protected render(): TemplateResult {
     return html`
       <style>
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            this._editorToggle && this._configElement !== null
+              ? html`<div class="element-editor">${when(
+                  this._configElement,
+                  () => this._configElement,
+                  () => html`Loading...`
+                )}</div>`
+              : html`
+              <hui-yaml-editor
+                .yaml="${this._currentConfigYaml}"
+                @yaml-changed="${this._handleYamlChanged}"
+              ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
-            .yaml="${this._cardConfig}"
+            .yaml="${this._currentConfigYaml}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
-  private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+  private _handleYamlChanged(ev: YamlChangedEvent): void {
+    this._configValue = { format: "yaml", value: ev.detail.yaml };
+    this._updatePreview(this._configValue);
+  }
+
+  private _handleConfigChanged(value: LovelaceConfig): void {
+    if (this._configElement) {
```

Comment:
Is it possible that this method is called and there is no config element?

Reply:
Shouldnt be since I set it in the beginning

Reply:
No...